compass/ocean/tests/parabolic_bowl/lts/lts_regions.py

```diff
@@ -0,0 +1,332 @@
+import math
+import os
+
+import netCDF4 as nc
+import numpy as np
+import xarray as xr
+from mpas_tools.io import write_netcdf
+from mpas_tools.viz.paraview_extractor import extract_vtk
+
+from compass.step import Step
+
+
+class LTSRegions(Step):
+    """
+    A step for adding LTS regions to a global MPAS-Ocean mesh
+
+    Attributes
+    ----------
+    initial_state_step :
+        compass.ocean.tests.dam_break.initial_state.InitialState
+        The initial step containing input files to this step
+    """
+    def __init__(self, test_case, init_step,
+                 name='lts_regions', subdir='lts_regions'):
+        """
+        Create a new step
+
+        Parameters
+        ----------
+        test_case : compass.Testcase
+            The test case this step belongs to
+
+        init_step :
+            compass.ocean.tests.dam_break.initial_state.InitialState
+            The initial state step containing input files to this step
+
+        name : str, optional
+            the name of the step
+
+        subdir : str, optional
+            the subdirectory for the step.  The default is ``name``
+        """
+        super().__init__(test_case, name=name, subdir=subdir)
+
+        for file in ['lts_mesh.nc', 'lts_graph.info', 'lts_ocean.nc']:
+            self.add_output_file(filename=file)
+
+        self.init_step = init_step
+
+    def setup(self):
+        """
+        Set up the test case in the work directory, including downloading any
+        dependencies.
+        """
+        super().setup()
+
+        init_path = self.init_step.path
+        tgt1 = os.path.join(init_path, 'culled_mesh.nc')
+        self.add_input_file(filename='culled_mesh.nc', work_dir_target=tgt1)
+
+        tgt2 = os.path.join(init_path, 'culled_graph.info')
+        self.add_input_file(filename='culled_graph.info', work_dir_target=tgt2)
+
+        tgt3 = os.path.join(init_path, 'ocean.nc')
+        self.add_input_file(filename='ocean.nc', work_dir_target=tgt3)
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+
+        use_progress_bar = self.log_filename is None
+        label_mesh(init='ocean.nc',
+                   mesh='culled_mesh.nc',
+                   graph_info='culled_graph.info', num_interface=2,
+                   logger=self.logger, use_progress_bar=use_progress_bar)
+
+
+def label_mesh(init, mesh, graph_info, num_interface,  # noqa: C901
```

Comment:
Is this routine the same as used in `dam_break`? If so, I think it would be best to migrate it to the ocean framework.

Reply:
Yes except for the way the fine regions are setup. Once they are defined, everything else is the same. We are making a change to this routine with Jeremy to accommodate something that we need for Forward-Backward LTS, so once we do that we could also add an option to specify separately how to define the fine regions and add the rest of the routine to the ocean framework. For now, let's leave it like it is if you don't mind.

Reply:
Ok. As long as it is on your radar to get migrated to the framework when you're happy with it.